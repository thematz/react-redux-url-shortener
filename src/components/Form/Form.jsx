// src/components/Form/index.jsx

import React, { Component } from 'react'
import { bindActionCreators } from 'redux'
import { connect } from 'react-redux'
import { shorten, updateUrl } from '../../actions'

class Form extends Component {
    handlePaste = e => {
        const url = e.clipboardData.getData('Text')
        this.props.shorten(url)
    }
    handleSubmit = e => {
        e.preventDefault()
        if (this.props.url) {
            this.props.shorten(this.props.url)
        }
    }
    handleValueChange = value => {
        this.props.updateUrl(value)
    }
    render() {
        return (
            <section className="section">
                <form className="container" onSubmit={(e) => this.handleSubmit(e)}>
                    <div className="columns">
                        <div className="field has-addons column is-8 is-offset-2">
                            <div className="control is-expanded">
                                <input 
                                    className="input"
                                    type="text" 
                                    placeholder="Past a link to shorten it" 
                                    value={this.props.url} 
                                    onChange={(e) => this.handleValueChange(e.target.value)} 
                                    onPaste={this.handlePaste}
                                />
                            </div>
                            <div className="control">
                                <button className="button is-primary" type="submit">Shorten</button>
                            </div>
                        </div>
                    </div>
                </form>
            </section>
        )
    }
}

const mapStateToProps = state => ({
    url: state.url
})
const mapDispatchToProps = dispatch => bindActionCreators({ 
    shorten,
    updateUrl
}, dispatch)

export default connect(
    mapStateToProps,
    mapDispatchToProps
)(Form)