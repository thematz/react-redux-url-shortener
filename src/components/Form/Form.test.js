import React from 'react'
import Form from './Form'
import store from '../../store'

import Enzyme, { shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'
import toJson from 'enzyme-to-json'
Enzyme.configure({ adapter: new Adapter() })

it('renders Form correctly', () => {
    const wrapper = shallow(
        <Form store={store} />,
    ).dive()
    expect(toJson(wrapper)).toMatchSnapshot()
})