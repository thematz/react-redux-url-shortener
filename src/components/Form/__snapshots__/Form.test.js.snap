// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Form correctly 1`] = `
<section
  className="section"
>
  <form
    className="container"
    onSubmit={[Function]}
  >
    <div
      className="columns"
    >
      <div
        className="field has-addons column is-8 is-offset-2"
      >
        <div
          className="control is-expanded"
        >
          <input
            className="input"
            onChange={[Function]}
            onPaste={[Function]}
            placeholder="Past a link to shorten it"
            type="text"
            value=""
          />
        </div>
        <div
          className="control"
        >
          <button
            className="button is-primary"
            type="submit"
          >
            Shorten
          </button>
        </div>
      </div>
    </div>
  </form>
</section>
`;
